src/components/PrivateRoute/PrivateRoute.tsx

```diff
@@ -5,7 +5,8 @@ import { PrivateRouteBase } from "./PrivateRouteBase";
 export const PrivateRoute = async ({ children }: PropsWithChildren) => {
   const user = await getServerUser();
 
-  if (!user) return null; // Prevent server side render of authorized page
+  // prevent server side render of authorized page
+  const view = !user ? null : children;
```

Comment:
```suggestion
  if(!user) redirect ('/login') 
```

Wdyt? @josephdburdick @jakubczarnowski

Reply:
This looks fine to me but if we went in this direction what purpose would `withPrivateRoute` serve?

Reply:
In some scenarios withPrivateRoute can be faster than PrivateRoute because it's a client component (The useQuery can be reloaded before this server component is recalculated)

Reply:
I understand the client could be faster but your proposal would short-circuit at the server level before the client side ever had a chance to run.

Or am I missing something? Perhaps it only short-circuits with a redirect upon navigating but if a user's authentication expires the client side withPrivateRoute would redirect from there?